tests/test_utils.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for avro2py/utils.py"""
+import datetime
+import uuid
+
+from hypothesis import strategies as st, given
+
+from avro2py.utils import to_avro_dict
+from example.marketprice.messages.wallyworld.bidder import (
+    WallyworldAdGroupStructure, WallyworldAdItem
+)
+
+
+@given(st.integers(), st.text(), st.decimals())
+def test_to_avro_dict_nested_named_tuples(random_int, item_id, bid):
+    requested_at = datetime.datetime.utcnow()
+    campaign_id = ad_group_id = advertiser_id = random_int
+    users_lined = [uuid.uuid4() for _ in range(3)]
+    payload = WallyworldAdGroupStructure(
+        advertiserId=advertiser_id,
+        campaignId=campaign_id,
+        adGroupId=ad_group_id,
+        state="Paused",
+        users_linked=users_lined,
+        adItems=[WallyworldAdItem(
+            advertiserId=advertiser_id,
+            campaignId=campaign_id,
+            adGroupId=ad_group_id,
+            adItemId=4,
+            itemId=item_id,
+            state="Enabled",
+            bid=bid,
+            requestedAt=requested_at
+        )],
+        requestedAt=requested_at
+    )
+    value = to_avro_dict(payload)
+    assert value == {
+        "advertiserId": advertiser_id,
+        "campaignId": advertiser_id,
+        "adGroupId": advertiser_id,
+        "state": "Paused",
+        "users_linked": users_lined,
+        "adItems": [
+            {
+                "advertiserId": advertiser_id,
+                "campaignId": advertiser_id,
+                "adGroupId": advertiser_id,
+                "adItemId": 4,
+                "itemId": item_id,
+                "state": "Enabled",
+                "bid": bid,
+                "requestedAt": requested_at
+            }
+        ],
+        "requestedAt": requested_at
+    }
```

Comment:
Without this fix, it would create
```python
{'advertiserId': 0, 'campaignId': 0, 'adGroupId': 0, 'state': 'Paused',
 'users_linked': [UUID('b5d2edfd-3e34-41d5-8fad-f7f2f9d9c39e'),
                  UUID('21d0f873-8165-47af-a077-b98fe2dfe446'),
                  UUID('ba62e803-d3b1-4781-a39e-9b8ff0363e57')], 'adItems': [
WallyworldAdItem(advertiserId=0, campaignId=0, adGroupId=0, adItemId=4, itemId='', state='Enabled',
                 bid=Decimal('NaN'), requestedAt=datetime.datetime(2021, 9, 24, 11, 5, 15, 986801))],
 'requestedAt': datetime.datetime(2021, 9, 24, 11, 5, 15, 986801)}
```